rpcutil/validate.go

```diff
@@ -205,3 +208,33 @@ func ValidateEmail(v interface{}, _ string) error {
 	}
 	return err
 }
+
+// ValidateFutureTime validates that the time is > than now + (duration value)
+// The default value is -15m (-15 minutes). It is not exactly 0 hours
+// to prevent client time differences causing user frustration. Do not rely on
+// the default value being -15m as it might change slightly in the future. If
+// you want 0 hours just set "0".
+// For example: futureTime=-4h validates that the time is > now - 4 hours
+func ValidateFutureTime(v interface{}, s string) error {
+	var err error
+	offset := -15 * time.Minute
+	if s != "" {
+		offset, err = time.ParseDuration(s)
+		if err != nil {
+			return validator.ErrBadParameter
+		}
+	}
+	success := true
+	switch t := v.(type) {
```

Comment:
if you wanna be slick I think you could do something like:

```go
t, ok := v.(interface{ After(time.Time) bool })
if !ok {
    return validator.ErrUnsupported
} else if !t.After(time.Now().Add(offset)) {
    return fmt.Errorf("time is not greater than now %s", s)
}
```

and that would support other types which implement After besides those two, if there ever are any other types which do that.